src/utils/subgraph.ts

```diff
@@ -426,26 +430,27 @@ export async function getHighestLiquidityDIDs(
   chainIds: number[]
 ): Promise<string> {
   const didList: string[] = []
+  let highestLiquidiyAssets: HighestLiquidiyAssetsPools[] = []
   for (const chain of chainIds) {
     const queryContext: OperationContext = {
       url: `${getSubgrahUri(
         Number(chain)
       )}/subgraphs/name/oceanprotocol/ocean-subgraph`,
       requestPolicy: 'network-only'
     }
-    const fetchedPools = await fetchData(
-      HighestLiquidityAssets,
-      null,
-      queryContext
+    const fetchedPools: OperationResult<HighestLiquidiyAssets, any> =
+      await fetchData(HighestLiquidityAssets, null, queryContext)
+    highestLiquidiyAssets = highestLiquidiyAssets.concat(
+      fetchedPools.data.pools
     )
-    if (fetchedPools.data?.pools?.length === 0) return null
-    for (let i = 0; i < fetchedPools.data.pools.length; i++) {
-      if (!fetchedPools.data.pools[i].datatokenAddress) continue
-      const did = web3.utils
-        .toChecksumAddress(fetchedPools.data.pools[i].datatokenAddress)
-        .replace('0x', 'did:op:')
-      didList.push(did)
-    }
+  }
+  highestLiquidiyAssets.sort((a, b) => a.valueLocked - b.valueLocked).reverse()
```

Comment:
Seems like this ordering leads to the correct result.

Reply:
tending towards sneaking in small change towards ordering by `oceanReserve` in here as we're refactoring and touching it. I'm undecided by now but current implementation indeed feels as only being technically correct, whereas the arguments for `oceanReserve` are all good points from user standpoint like @w1kke & @DMats brought up and we generally tend towards user perspective. And realistically it should only lead to noticeable differences between new and very old pools.

Reply:
sure we can try ordering by oceanReserve, i tried to reproduce the actual order in our live market where we sort asset by valueLocked see https://github.com/oceanprotocol/market/blob/main/src/utils/subgraph.ts#L127-L128.
I'm gonna push the changes so we can have a look and decide if we stick to order by valueLocked or go with oceanReserve